source/groupBy.js

```diff
@@ -0,0 +1,30 @@
+'use strict'
+
+/**
+ * Функция, группирующая объекты из массива по типу
+ * @param {Array<Objects>} input - массив объектов
+ * @param key - ключ, по которому группируются элементы массива 
+ * 
+ * @returns {type: {Objects}}
+ */
+const groupBy = (input, key) => {
+    if (typeof(key) != 'string') {
+        key = 'category';
+    }
+    if (!Array.isArray(input)) {
+        throw new TypeError('Первый аргумент должен быть массивом объектов');
+    }
+    const itemsByCategory = input.reduce(function(result, object) {
+        if (!result[object[key]]) {
+            result[object[key]] = [object];
+        } else {
+            result[object[key]].push(object);
+        }
+        return result;
```

Comment:
Давай еще вынесем значение и разделим на логические блоки, чтобы облегчить чтение кода
```suggestion
        const category = object[key];
        
        if (!result[category]) {
            result[category] = [object];
        } else {
            result[category].push(object);
        }
        
        return result;
```